.github/workflows/reusable-plugin-size-audit.yml

```diff
@@ -0,0 +1,299 @@
+name: Reusable Audit Plugin Size
+
+on:
+  workflow_call:
+    inputs:
+      plugin-repo:
+        description: The plugin repo (newfold-labs/wp-plugin-name).
+        type: string
+        required: true
+      plugin-target-branch:
+        description: The target (base) branch of the plugin. This is used as the target branch for the pull request.
+        type: string
+        required: true
+      plugin-source-branch:
+        description: The source (head) branch of the plugin. This is used as a starting point for the version changes. For a normal release it would be develop, but for a hotfix it would be the hotfixing branch.
+        type: string
+        required: true
+env:
+  MAX_SIZE_INCREASE_PERCENT: 10
+  MAX_ZIP_SIZE_MB: 10
+
+jobs:
+  size-audit:
+    name: Audit Plugin Size
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          repository: ${{ inputs.plugin-repo }}
+          ref: ${{ inputs.plugin-source-branch }}
+          persist-credentials: true
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@bf6b4fbd49ca58e4608c9c89fba0b8d90bd2a39f # v2.35.5
+        with:
+          php-version: '8.1'
+          coverage: none
+          tools: composer, cs2pr
+
+      - name: Setup workflow context
+        id: workflow
+        working-directory: ${{ runner.temp }}
+        env:
+          REPO: ${{ inputs.plugin-repo }}
+          PLUGIN_NAME: ${{ vars.PLUGIN_NAME }}
+        run: |
+          mkdir dist
+          echo "DIST=${PWD}/dist" >> "$GITHUB_OUTPUT"
+          echo "PACKAGE=$PLUGIN_NAME" >> "$GITHUB_OUTPUT"
+
+      - name: Use Node.js 24.x
+        uses: actions/setup-node@2028fbc5c25fe9cf00d9f06a71cc4710d4507903 # v6.0.0
+        with:
+          node-version: 24.x
```

Comment:
Every repository should contain an `.nvmrc` file. I started the effort to ensure this was the case for all repositories under the organization, but haven't got past the module repositories (see [PRESS0-2908](https://jira.newfold.com/browse/PRESS0-2908)/[PRESS0-2909](https://jira.newfold.com/browse/PRESS0-2909)).

Different tools and repositories may have different Node.js/npm version requirements, so relying on that file through `node-version-file` input is preferred.